BitFlow/Eval/TorchEval.py

```diff
@@ -21,8 +21,25 @@ def eval_Round(self, a, prec, node: DagNode):
         scale = 2.0**prec
         return IntRound(a * scale)/scale
 
+    def eval_Relu(self, a, node: DagNode):
+            if a > 0:
+                return a
+            else:
+                return 0
+
     def eval_Select(self, a, node: DagNode):
+
         if len(a.shape) == 1:
             return a[node.index]
         else:
             return a[:, node.index]
+
+    def eval_Reduce(self, a, node: DagNode):
```

Comment:
Could you add a parameter to the Reduce DagNode that takes in the dimension you want to sum over? You should be able to just use torch.sum(a, dim=node.reduce_dim) for the evaluation function. Could you also write a couple tests testing reducing over different dimensions?